deployment/postgres-init-db/sql/views/merlin/derived_events.sql

```diff
@@ -0,0 +1,106 @@
+-- Rule 1. An External Event superseded by nothing will be present in the final, derived result.
+-- Rule 2. An External Event partially superseded by a later External Source, but whose start time occurs before the start of said External Source(s), will be present in the final, derived result.
+-- Rule 3. An External Event whose start is superseded by another External Source, even if its end occurs after the end of said External Source, will be replaced by the contents of that External Source (whether they are blank spaces, or other events).
+-- Rule 4. An External Event who shares an ID with an External Event in a later External Source will always be replaced.
+
+create materialized view merlin.derived_events as
+-- "distinct on (event_key, derivation_group_name)" and "order by valid_at" satisfies rule 4
+-- (only the most recently valid version of an event is included)
+select distinct on (event_key, derivation_group_name)
+    output.event_key,
+    output.source_key,
+    output.derivation_group_name,
+    output.event_type_name,
+    output.duration,
+    output.start_time,
+    output.source_range,
+    output.valid_at
+from (
+  -- select the events from the sources and include them as they fit into the ranges determined by sub
+  select
+    s.key as source_key,
+    ee.key as event_key,
+    ee.event_type_name,
+    ee.duration,
+    s.derivation_group_name,
+    ee.start_time,
+    s.source_range,
+    s.valid_at
+  from merlin.external_event ee
+  join (
+    with base_ranges as (
+      -- base_ranges orders sources by their valid time
+      -- and extracts the multirange that they are stated to be valid over
+      select
+        external_source.key,
+        external_source.derivation_group_name,
+        tstzmultirange(tstzrange(external_source.start_time, external_source.end_time)) as range,
+        external_source.valid_at
+      from merlin.external_source
+      order by external_source.valid_at
+    ), base_and_sub_ranges as (
+      -- base_and_sub_ranges takes each of the sources above and compiles a list of all the sources that follow it
+      -- and their multiranges that they are stated to be valid over
+      select
+        base.key,
+        base.derivation_group_name,
+        base.range                                                      as original_range,
```

Comment:
Somehow a lot of spaces got inserted between `base.range` and `as`. Please remove them.
```suggestion
        base.range as original_range,
```